File: 101/exercise_rework/easy_6/halvsies.rb
# halvsies.rb

def halvsies(arr)
  input_length = arr.length
  half = arr.length / 2
  if input_length.odd?
    first_arr = arr.shift(half + 1)
  else
    first_arr = arr.shift(half)
  end
  [first_arr, arr]
end

File: 210/js_small_problems/easy_4/halvsies.js
// halvsies.js

function halvsies(arr) {
  var half;
  var firstHalf;
  var secondHalf;
  half = arr.length / 2
  if (arr.length % 2 === 1) { half += 1 }
  firstHalf = arr.slice(0, half);
  secondHalf = arr.slice(half, arr.length)
  return [ firstHalf, secondHalf ];
}

console.log(halvsies([1, 2, 3, 4]));       // [[1, 2], [3, 4]]
console.log(halvsies([1, 5, 2, 4, 3]));    // [[1, 5, 2], [4, 3]]
console.log(halvsies([5]));                // [[5], []]
console.log(halvsies([]));                 // [[], []]